### DISCUSSION TITLE
Why is pylance not picking up my custom package

### DISCUSSION THREAD

**Discussion by Riprock** [microsoft/pylance-release#6568] [Q&A] [answered]
Hello Guys. I created a custom package and I did the needful of making a venv, enabling it, installing the package with just a pip install and yet pylance errors out that it cant find it. Its only when I create a .vscode/settings.json with
```
{
    "python.analysis.extraPaths": ["./"]
}
```
That pylance knows that its a legitamate package. The venv is activated in the terminal and set as the interpreter for vscode. 
https://github.com/Riprock/acintellisense/tree/main - This is my project that im having the issues with
What am I doing wrong?

**Comment by rchiodo** [comment]
How did you install it? Editable installs that use dynamic code generally don't work. See this documentation:

https://microsoft.github.io/pyright/#/import-resolution?id=editable-installs

**Reply by Riprock** [reply]
I created a new space. Did the following
```
python3 -m venv venv
source ./ven/bin/activate
pip install acintellisense
```

When I run a file calling from this package from the terminal it work. Its just pylance is having a hard time finding it

**Reply by rchiodo** [reply]
It looks to me like the installer didn't install anything:

![image](https://github.com/user-attachments/assets/149254f9-aeac-4b3d-a37e-3b228d89a06b)


After pip installing it, I can't use it:

```
PS C:\Users\rchiodo\source\testing\test_pylance> python -m pip install acintellisense            
Collecting acintellisense
  Downloading acintellisense-1.1.1-py3-none-any.whl.metadata (2.7 kB)
Downloading acintellisense-1.1.1-py3-none-any.whl (17 kB)
Installing collected packages: acintellisense
Successfully installed acintellisense-1.1.1
PS C:\Users\rchiodo\source\testing\test_pylance> python
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import acintellisense
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'acintellisense'
>>>
```

**Reply by Riprock** [reply]
import ac

Thats the module name that all the functions are contained in

**Reply by Riprock** [reply]
```venv) riprock@shadowbox:~/Downloads/ASenseTesting$ python
Python 3.12.7 (main, Oct  3 2024, 15:15:22) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ac
>>> import acintellisense
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'acintellisense'
>>> 
```

**Reply by Riprock** [reply]
For context this is a stub documentation package for making mods for a racing simulator called assetto corsa. ac is the main package that gets loaded by the python interpreter that the game loads. The goal is to make where when writing the plugins, you can import ac, so when its copied devs dont have to change the import for any reason. They just load the files as is.

**Reply by rchiodo** [reply, marked as answer]
Then that works for me. 

![image](https://github.com/user-attachments/assets/9b2a6926-6ecc-4bc5-b367-14fa7821d9f6)

If you upload your log as described here it should say why we couldn't find the `ac` module:
https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue

**Reply by Riprock** [reply]
Okay this is weird. I just closed out of all of my open vscode windows(Diffrent projects). Restarted my PC and deleted the settings.json and it figured it out. No idea what was wrong in the first place but thank you for reaching out and helping! I really appreciate it!
